### PR TITLE
remove bcl converter from CLI

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Removal of bcl2fastq option from CLI commands:
```shell
cg demultiplex samplesheet create
cg demultiplex flow-cell
cg demultiplex finish flow-cell
```

### Changed

- Parameter `bcl_converter` od sample sheet API function `get_or_create_sample_sheet` from mandatory to optional set to `None` by default.

### Removed

- `@OPTION_BCL_CONVERTER` decorator
- `bcl_converter` parameter from all CLI commands
- CLI test functions

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b remove-bcl2fastq-commands -a
    ```

### How to test and expected outcomes

- [x] Print help from commands, verify that the option is no longer there
- [x] Try using them with the `-b bcl2fastq` flag, see that an error is raised
- [x] Verify that the commands work with BCLConvert when used without the flag

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MINOR** - when you add functionality in a backwards compatible manner

## Implementation Plan

- [ ] Document in Atlas
- [ ] Deploy this branch on 
  - [ ] cg stage
  - [ ] cg production 
- [ ] Inform to @Clinical-Genomics/prodbioinfo

### CODE REVIEW

File: cg/apps/demultiplex/sample_sheet/api.py [hunk 1]
```diff
@@ -272,7 +272,9 @@ def _create_sample_sheet_file(self, flow_cell: FlowCellDirectoryData) -> None:
             flow_cell_directory=flow_cell.path, flow_cell_name=flow_cell.id, hk_api=self.hk_api
         )
 
-    def get_or_create_sample_sheet(self, flow_cell_name: str, bcl_converter: str | None) -> None:
+    def get_or_create_sample_sheet(
+        self, flow_cell_name: str, bcl_converter: str | None = None
```
Comment (diitaz93) [inline cg/apps/demultiplex/sample_sheet/api.py:276]: removal of bcl_converter from this function is put of scope. Setting it to `None` for now

File: cg/cli/demultiplex/demux.py [hunk 2]
```diff
@@ -80,20 +80,18 @@ def demultiplex_all(context: CGConfig, flow_cells_directory: click.Path, dry_run
...
-@OPTION_BCL_CONVERTER
 @click.pass_obj
 def demultiplex_flow_cell(
     context: CGConfig,
     dry_run: bool,
     flow_cell_name: str,
-    bcl_converter: str,
 ):
-    """Demultiplex a flow cell.
+    """Demultiplex a flow cell using BclConverter.
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/demux.py:89]:
```suggestion
    """Demultiplex a flow cell using BCLConvert.
```
"BCL Convert" is the software name

**Review by ChrOertlin — APPROVED** [review]
Looks good to me. Are thre some fixtures we need/can remove?

**Comment by diitaz93** [comment]
## Tests on stage
Printing help
```shell
$ cg demultiplex samplesheet create --help
Running cg demultiplex.
Usage: cg demultiplex samplesheet create [OPTIONS] FLOW_CELL_NAME

  Create a sample sheet or hard-link it from Housekeeper in the flow cell
  directory.

  'flow-cell-name' is the flow cell run directory name, e.g.
  '181005_D00410_0735_BHM2LNBCX2'

  Search the flow cell in the directory specified in config.

Options:
  -d, --dry-run  Runs the command without making any changes
  --force        Skips the validation of the sample sheet
  --help         Show this message and exit.
[js.diazboada@hasta:~] [S_base] 24s $ cg demultiplex flow-cell --help
Running cg demultiplex.
Usage: cg demultiplex flow-cell [OPTIONS] FLOW_CELL_NAME

  Demultiplex a flow cell using BclConverter.

  flow cell name is the flow cell run directory name, e.g.
  '201203_D00483_0200_AHVKJCDRXX'

Options:
  --dry-run
  --help     Show this message and exit.
[js.diazboada@hasta:~] [S_base] 23s $ cg demultiplex finish flow-cell --help
Running cg demultiplex.
Usage: cg demultiplex finish flow-cell [OPTIONS] FLOW_CELL_DIRECTORY_NAME

  Command to finish up a flow cell after demultiplexing.

  flow-cell-name is full flow cell name, e.g. '201203_D00483_0200_AHVKJCDRXX'.

Options:
  --force
  -d, --dry-run  Runs the command without making any changes
  --help         Show this message and exit.
[js.diazboada@hasta:~] [S_base] 22s $ 
```
Using option `-b` fails
```shell
$ cg demultiplex samplesheet create 180509_D00450_0598_BHGYFNBCX2 -b bcl2fastq
Running cg demultiplex.
Usage: cg demultiplex samplesheet create [OPTIONS] FLOW_CELL_NAME
Try 'cg demultiplex samplesheet create --help' for help.

Error: No such option: -b
[js.diazboada@hasta:/home/proj/stage/sequencing_data/illumina/flow_cells] [S_base] 22s 2 $ cg demultiplex flow-cell 180509_D00450_0598_BHGYFNBCX2 -b bcl2fastq
Running cg demultiplex.
Usage: cg demultiplex flow-cell [OPTIONS] FLOW_CELL_NAME
Try 'cg demultiplex flow-cell --help' for help.

Error: No such option: -b
[js.diazboada@hasta:/home/proj/stage/sequencing_data/illumina/flow_cells] [S_base] 23s 2 $ cg demultiplex finish flow-cell 180509_D00450_0598_BHGYFNBCX2 -b bcl2fastq
Running cg demultiplex.
Usage: cg demultiplex finish flow-cell [OPTIONS] FLOW_CELL_DIRECTORY_NAME
Try 'cg demultiplex finish flow-cell --help' for help.

Error: No such option: -b
[js.diazboada@hasta:/home/proj/stage/sequencing_data/illumina/flow_cells] [S_base] 21s 2 $ 
```

**Comment by diitaz93** [comment]
> Looks good to me. Are thre some fixtures we need/can remove?

Oh yes, many. This will be taken care of in the next PRs (I already removed some)

**Comment by diitaz93** [comment]
## Deployed to stage:
```shell
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... 
done.
Log deploy... done.
cg, version 59.24.0
[js.diazboada@hasta:] [S_base] 3m11s $ up
```
## Deployed to production:
```shell
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... 
done.
Log deploy... done.
cg, version 59.24.0
θ60° [js.diazboada@hasta:] [P_base] 2m39s $
```

**Pull request merged**